### PR TITLE
Reg front end copy2

### PR DESCRIPTION
**Pull request by chowkingman**

### CODE REVIEW

File: server/server.ts [hunk 1]
```diff
@@ -103,4 +103,29 @@ app.get("/api/generateDaily",(req, res) => {
     new dailyEntriesModel(daily).save();
   }
   res.send({message:"generated 100 datas!"});
-})
\ No newline at end of file
+})
+
+//Below is a post request for the users to register
+app.post("/api/register", (req, res) => {
```
Comment (RetroInferno) [inline server/server.ts:109]: need to declare this as async and use: 
```
try {
  // some code........
  await userReg.save();
  res.sendStatus(201);
} catch (err) {
  res.status(501);
  res.send(err);
```

File: server/server.ts [hunk 2]
```diff
@@ -103,4 +103,29 @@ app.get("/api/generateDaily",(req, res) => {
...
+    email: userData.email,
+    phoneNumber: +userData.phoneNumber,
+    height: +userData.height,
+    weight: +userData.weight,
+    caloriesGoal: +userData.targetCalories,
+    waterGoal: +userData.targetWater,
+    createdTime: new Date(),
+    updatedTime:{},
+  });
+  //userReg.save();
```
Comment (RetroInferno) [inline server/server.ts:126]: Why wouldn't we save?

File: server/server.ts [hunk 3]
```diff
@@ -103,4 +103,29 @@ app.get("/api/generateDaily",(req, res) => {
...
+  res.send({
+    code: 200,
+    essmsg: "user successfully post a request"
+  });
```
Comment (RetroInferno) [inline server/server.ts:130]: See Async example. Need to use stand Http codes for standard error handling

File: server/server.ts [hunk 4]
```diff
@@ -103,4 +103,29 @@ app.get("/api/generateDaily",(req, res) => {
...
+    age: +userData.age,
+    gender: userData.gender.value,
+    email: userData.email,
+    phoneNumber: +userData.phoneNumber,
+    height: +userData.height,
+    weight: +userData.weight,
+    caloriesGoal: +userData.targetCalories,
+    waterGoal: +userData.targetWater,
+    createdTime: new Date(),
+    updatedTime:{},
```
Comment (RetroInferno) [inline server/server.ts:124]: Don't need the `updatedTime: {}`. This will be automatically done by mongoose.

File: server/server.ts [hunk 5]
```diff
@@ -103,4 +103,29 @@ app.get("/api/generateDaily",(req, res) => {
...
+  let userReg = new userEntriesModel({
+    firstName: userData.firstName,
+    lastName: userData.lastName,
+    age: +userData.age,
+    gender: userData.gender.value,
+    email: userData.email,
+    phoneNumber: +userData.phoneNumber,
+    height: +userData.height,
+    weight: +userData.weight,
+    caloriesGoal: +userData.targetCalories,
```
Comment (RetroInferno) [inline server/server.ts:121]: would we want some to use a suggested calorie goal if this is not provided or is it on the front end?

**Review by RetroInferno — CHANGES_REQUESTED** [review]

File: server/server.ts (hunk 1 above)
Comment (chowkingman) [reply server/server.ts:109]: Good call, I missed it.

File: server/server.ts (hunk 3 above)
Comment (chowkingman) [reply server/server.ts:130]: updated.

File: server/server.ts (hunk 4 above)
Comment (chowkingman) [reply server/server.ts:124]: good call.

File: server/server.ts (hunk 2 above)
Comment (chowkingman) [reply server/server.ts:126]: just allowed saving, hopefully, it will not destroy the Atlas DB.

File: server/server.ts (hunk 5 above)
Comment (chowkingman) [reply server/server.ts:121]: Good call! On server.ts and the userSchema, I added caloriesRecommanded. I will be doing the math on the front end to figure out the recommended calories based on the BMI, and then I will send the number to the database to store.

**Pull request closed without merging**